src/js/pages/jobs/JobsTab.js

```diff
@@ -81,11 +72,17 @@ class JobsTab extends mixin(StoreMixin, QueryParamsMixin, SaveStateMixin) {
     this.setState({isJobFormModalOpen: true});
   }
 
-  handleFilterChange(filterValues, filterType) {
-    var stateChanges = Object.assign({}, this.state);
-    stateChanges[filterType] = filterValues;
+  handleFilterChange(filterValue) {
+    const {router} = this.context;
+    const {location: {pathname}} = this.props;
+    const query = {};
```

Comment:
nit: You could do it right here `const query = {[JobFilterTypes.TEXT]: filterValue};`